### DISCUSSION TITLE
Third party cookie deprecation (Privacy Sandbox)

### DISCUSSION THREAD

**Discussion by joselcvarela** [directus/directus#21913] [Feature Requests]
## Description
During 2024, Chrome will be removing support for third party cookies in order to improve privacy.
More information can be seen here: https://privacysandbox.com/

This change will impact all users that have Directus hosted in a different top domain than the  main website and:
- are using `mode: cookie` on login endpoint
- or are using `REFRESH_TOKEN_COOKIE_SAME_SITE=none` and logging in from another domain. For example, using [Seamless SSO](https://docs.directus.io/self-hosted/sso.html#seamless-sso)

Right now, Chrome users are already receiving a warning "Third-party cookie will be blocked. Learn more in the Issues tab."

More information about SSO cookies and privacy sandbox can be seen here:
https://developers.google.com/privacy-sandbox/3pcd/guides/identity#sign-in_solutions

## Solutions
- Have users putting Directus in the same domain as the main website. For example, main website is `example.com` so if users put Directus under `directus.example.com` no more changes may need to happen.
This solution does not require any changes in Directus

- Use Related Website Set: https://developers.google.com/privacy-sandbox/3pcd/related-website-sets-integration
This solution seems to not require any changes in Directus

- Use FedCM: https://developers.google.com/privacy-sandbox/3pcd/fedcm-developer-guide
This solution requires some changes to Directus

### Motivation

Basically Google Chrome is removing support for third party cookies.

### Detailed Design

The solution that requires some changes on Directus is FedCM and the implementation can be seen here:
https://developers.google.com/privacy-sandbox/3pcd/fedcm-developer-guide

### Must Have:
- FedCM support?

### Drawbacks

- It seems to only affect Chrome users
- Only for users that are using Directus cookies across domains

### Alternatives

- Have Directus under the same domain as the main website
- Use [Related Website Set](https://developers.google.com/privacy-sandbox/3pcd/related-website-sets-integration)

### Adoption Strategy

N/A

**Comment by br41nslug** [comment]
Hmmm I would be somewhat hesitant to implement a new authentication provider that just works for Google's Chrome browser 🤔 My preference here would be to wait for FedCM to become a standard which is implemented by the other browsers (as it could be their implementations may not be 100% spec compliant)

**Comment by Zehir** [comment]
There is also this solution I proposed for that issue ; https://github.com/directus/directus/discussions/20548

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
